### PR TITLE
Make test describe statements consistent

### PR DESCRIPTION
**Pull request by baer**
This PR does 3 things:
1. Correct bad describe path in pluggable spec
2. Move construct spec to mirror src directory
3. Remove nested describe in resolve spec to make it consistent with other tests

One other thing I'd like to do is to rename `construct.spec.js` to be `index.spec.js` and to remove the unnecessary `describe("common module" ...)`. For those two since it is a preference and not a consistency choice I thought I'd ask your thoughts first. :)

### CODE REVIEW

File: spec/lib/resolve.spec.js [hunk 1]
```diff
@@ -5,60 +5,58 @@ var resolve = require("../../lib/resolve");
 var baseDir = path.resolve(__dirname, "../..");
 
 
-describe("lib/", function () {
-  describe("resolve", function () {
```
Comment (baer) [inline spec/lib/resolve.spec.js]: I removed this inner describe and made the top level `lib/resolve` to make it similar to the other tests. The rest is just whitespace.

**Comment by divmain** [comment]
This moves `spec/lib/construct.spec.js` → `spec/lib/compile/construct/construct.spec.js`.  But by my way of thinking, this spec file tests the whole module.  So for now, the file should be `spec/lib/compile/construct.spec.js`.  If we determine to split it out into multiple spec files, then we can add a subdir.  In other words, I'm treating this just as the `require` statements treat it.  Thoughts?

File: spec/lib/resolve.spec.js (hunk 1 above)
Comment (divmain) [inline spec/lib/resolve.spec.js]: :+1:

**Comment by baer** [comment]
To me, it seems that in keeping it one dir back you're breaking the 1:1 mapping between src and test that I've come to expect in projects. That said, I see your point since `index.js` is testable in it's own right.

Maybe the right answer here is a delineation between (or at least a convention for) unit vs. integration tests.

**Comment by baer** [comment]
Also, this is 100% bikeshedding so I am going to hereby nominate you BDFL and say that whatever your answer is should be the way it is. Any further discussion is just a distraction. This PR was just be being fastidious as I read things :)

**Comment by divmain** [comment]
Meh, I don't mind the conversation :).  The line between unit and integration tests is sort of fuzzy for me, because ultimately it really depends on how you define your unit.

My preference here is probably influenced by the fact that the 1-to-1 mapping of src-to-spec that is very common is fundamentally influenced by implementation.  In the spec, we're requiring in `"../../lib/compile/construct"`.  The fact that that is a directory module instead of a file module is an implementation detail, and everything else about the tests remain the same.

If, for example, `pluggable.js` got larger and it was changed into a directory module, the tests wouldn't have to change, and I suppose I see that as a good thing.  It is only a bad thing if the test suite gets too large or too unwieldy and you end up with unrelated tests in the same file.

That said, as I mentioned, that's sort of a fuzzy metric to use.  My preference for now is to have the spec file mirror the directory module.  But if there's a good reason to move it or split it up down the line, we can do that.

**Comment by baer** [comment]
Sounds good to me. Done!

**Comment by divmain** [comment]
Cool beans.  Thanks!

**Pull request merged**